zeebe/stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java

```diff
@@ -407,10 +409,18 @@ public void shouldProcessFollowUpEventsAndCommands() {
     assertThat(firstRecord.getSourceEventPosition()).isEqualTo(-1);
     final var firstRecordPosition = firstRecord.getPosition();
 
+    final var firstRecordMetadata = new RecordMetadata();
+    firstRecord.readMetadata(firstRecordMetadata);
+    assertThat(firstRecordMetadata.getRequestId()).isEqualTo(requestId);
+
     await("should write follow up events")
         .untilAsserted(() -> assertThat(logStreamReader.hasNext()).isTrue());
     while (logStreamReader.hasNext()) {
-      assertThat(logStreamReader.next().getSourceEventPosition()).isEqualTo(firstRecordPosition);
+      final LoggedEvent followup = logStreamReader.next();
+      assertThat(followup.getSourceEventPosition()).isEqualTo(firstRecordPosition);
+      final var followupMetadata = new RecordMetadata();
+      followup.readMetadata(followupMetadata);
+      assertThat(followupMetadata.getRequestId()).isEqualTo(requestId);
     }
   }
 
```

Comment:
@Zelldon just out of curiosity, asking since you were the author.
Is there any validation that the below test `shouldSetSourcePointerForFollowUpRecords` asserts and not covered by this above one (`shouldProcessFollowUpEventsAndCommands `)?

Reply:
I think the purpose/scope of both tests were different. But yeah based on the diff they look quite similar 🤔 I haven't check it in detail (and have no time for it yet) We could reduce the assertions in the first one, to reduce the overlap if you like and concentrate on the purpose of verifying that follow up commands are processed as well, I guess.

Reply:
Well, yeah I guess this a controversial topic I'm not sure whether I can discuss this here right now :D There are several use cases and benefits of repeating yourself in the matter of test cases. Sometimes you have to sacrifice one against another. For example, this allows us to highlight better where the issue lies and makes sure that one tests does one thing (SRP) as with one method which does many assertsion/verifications it is often hard to tell.

Reply:
But as I don't see currently an easy way of how to separate this, I'm find if you delete it.

Reply:
I removed it, and enhanced the asserts with descriptions above! Thank you 🙇‍♂️